### PR TITLE
[Fix-11616] Workflow Definition Name Change Validation Optimization.

### LINKED ISSUE #11616: [Bug] Modify workflow definition name validation error
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Modify workflow definition name validation error

### What you expected to happen

Normal

### How to reproduce

Change the existed workflow name.

### Anything else

No

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by WangJPLeo**
## Purpose of the pull request

Workflow Definition Name Change Validation Optimization.

## Brief change log

close #11616 

## Verify this pull request

Manually verified the change by testing locally.

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessDefinitionService.java [hunk 1]
```diff
@@ -194,11 +194,13 @@ Map<String, Object> updateProcessDefinition(User loginUser,
...
      * @param projectCode project code
      * @param name name
+     * @param code code
      * @return true if process definition name not exists, otherwise false
      */
     Map<String, Object> verifyProcessDefinitionName(User loginUser,
                                                     long projectCode,
-                                                    String name);
+                                                    String name,
+                                                    long code);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessDefinitionService.java]:
```suggestion
                                                    Long processDefinitionCode);
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java [hunk 2]
```diff
@@ -202,16 +202,18 @@ public Result moveProcessDefinition(@ApiIgnore @RequestAttribute(value = Constan
...
     @GetMapping(value = "/verify-name")
     @ResponseStatus(HttpStatus.OK)
     @ApiException(VERIFY_PROCESS_DEFINITION_NAME_UNIQUE_ERROR)
     @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
     public Result verifyProcessDefinitionName(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
                                               @ApiParam(name = "projectCode", value = "PROJECT_CODE", required = true) @PathVariable long projectCode,
-                                              @RequestParam(value = "name", required = true) String name) {
-        Map<String, Object> result = processDefinitionService.verifyProcessDefinitionName(loginUser, projectCode, name);
+                                              @RequestParam(value = "name", required = true) String name,
+                                              @RequestParam(value = "code", required = false, defaultValue = "0") long code) {
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java]:
```suggestion
                                              @RequestParam(value = "code", required = false) Long code) {
```

**Review by zhongjiajie — DISMISSED** [review]
LGTM

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java (hunk 2 above)
Comment (zhongjiajie) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java]: why should we remove the default value?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessDefinitionService.java (hunk 1 above)
Comment (WangJPLeo) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessDefinitionService.java]: ok.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java (hunk 2 above)
Comment (caishunfeng) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java]: > why should we remove the default value?

If use Long type, default is Null.

Comment (WangJPLeo) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java]: It is best to keep this default value. When updating, the front-end will pass empty, and the back-end will judge according to whether the code is 0.

**Review by caishunfeng — DISMISSED** [review]

**Review by zhongjiajie — APPROVED** [review]
LGTM

**Review by lenboo — APPROVED** [review]

**Pull request merged**